core/src/main/java/hudson/Functions.java

```diff
@@ -2289,4 +2290,59 @@ public static boolean isContextMenuVisible(Action a) {
             return true;
         }
     }
+
+    public static Icon tryGetIcon(String iconGuess) {
+        // Jenkins Symbols don't have metadata so return null
+        if (iconGuess == null || iconGuess.startsWith("symbol-")) {
+            return null;
+        }
+
+        StaplerRequest currentRequest = Stapler.getCurrentRequest();
+        currentRequest.getWebApp().getDispatchValidator().allowDispatch(currentRequest, Stapler.getCurrentResponse());
+        Icon iconMetadata = IconSet.icons.getIconByClassSpec(iconGuess);
+
+        if (iconMetadata == null) {
+            // Icon could be provided as a simple iconFileName e.g. "settings.png"
+            iconMetadata = IconSet.icons.getIconByClassSpec(IconSet.toNormalizedIconNameClass(iconGuess) + " icon-md");
+        }
+
+        if (iconMetadata == null) {
+            // Icon could be provided as an absolute iconFileName e.g. "/plugin/foo/abc.png"
+            iconMetadata = IconSet.icons.getIconByUrl(iconGuess);
+        }
+
+        return iconMetadata;
+    }
+
+    public static String tryGetIconPath(String iconGuess, JellyContext context) {
+        if (iconGuess == null) {
+            return null;
+        }
+
+        if (iconGuess.startsWith("symbol-")) {
+            return iconGuess;
+        }
+
+        StaplerRequest currentRequest = Stapler.getCurrentRequest();
+        currentRequest.getWebApp().getDispatchValidator().allowDispatch(currentRequest, Stapler.getCurrentResponse());
+        String rootURL = currentRequest.getContextPath();
+        Icon iconMetadata = tryGetIcon(iconGuess);
+        String iconSource = null;
+
+        if (iconMetadata != null) {
+            iconSource = iconMetadata.getQualifiedUrl(context);
+        }
+
+        if (iconMetadata == null) {
+            iconSource = rootURL + (iconGuess.startsWith("/images/") || iconGuess.startsWith("/plugin/") ? getResourcePath() : "") + getResourcePath() + iconGuess;
+        }
+
+        if (iconMetadata != null && iconMetadata.getClassSpec() != null) {
+            if (IconSet.tryTranslateTangoIconToSymbol(iconMetadata.getClassSpec()) != null) {
+                return IconSet.tryTranslateTangoIconToSymbol(iconMetadata.getClassSpec());
+            }
```

Comment:
no need to do this twice
```suggestion
            String icon = IconSet.tryTranslateTangoIconToSymbol(iconMetadata.getClassSpec());
            if (icon != null) {
                return icon;
            }
```